lib/include/types/iot_platform_types.h

```diff
@@ -30,6 +30,9 @@
 /* The config header is always included first. */
 #include "iot_config.h"
 
+/* Linear containers (lists and queues) include for metrics types. */
+#include "iot_linear_containers.h"
```

Comment:
Is this the right spot for this header?  (same question for metrics types down below...)

Reply:
The type `IotLink_t` is needed for the metrics structure.